shared/macros-oval.jinja

```diff
@@ -0,0 +1,88 @@
+{{%- macro oval_check_config_file(msg='', path='', prefix_regex='^\s*', parameter='', separator_regex='(?-i)[\s]+', value='', default='', application='', exact_match='') -%}}
+{{%- if default %}}
+{{%- set none_exist = "_none_exist" -%}}
+{{%- endif %}}
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    <metadata>
+      <title>{{{ msg or rule_title }}}</title>
+      <affected family="unix">
+        <platform>multi_platform_all</platform>
+      </affected>
+      <description>{{{ msg or rule_title }}}</description>
+    </metadata>
+    <criteria comment="{{{ application }}} is configured correctly"
+    operator="OR">
+        {{{ oval_line_in_file_criteria(path, parameter) }}}
+        {{%- if none_exist %}}
+        {{{ oval_line_in_file_criteria(path, parameter, none_exist) }}}
+        {{%- endif %}}
+    </criteria>
+  </definition>
+
+  {{{ oval_line_in_file_test(path, parameter) }}}
+  {{{ oval_line_in_file_object(path, prefix_regex, parameter, separator_regex, value) }}}
+  {{{ oval_line_in_file_state(value, exact_match) }}}
+  {{%- if none_exist %}}
+  {{{ oval_line_in_file_test(path, parameter, none_exist) }}}
+  {{{ oval_line_in_file_object(path, prefix_regex, parameter, separator_regex, value, none_exist) }}}
+  {{%- endif %}}
+
+</def-group>
+{{%- endmacro %}}
+
+{{%- macro oval_line_in_file_criteria(path='', parameter='', none_exist='') -%}}
+{{%- if none_exist %}}
+<criterion comment="Check absence of {{{ parameter }}} in {{{ path }}}"
+{{%- else %}}
+<criterion comment="Check {{{ parameter }}} in {{{ path }}}"
+{{%- endif %}}
+  test_ref="test_{{{ rule_id }}}{{{ none_exist }}}" />
+{{%- endmacro %}}
+
+{{%- macro oval_line_in_file_test(path='', parameter='', none_exist='') -%}}
+  {{%- if none_exist %}}
+  <ind:textfilecontent54_test check="all" check_existence="none_exist"
+  comment="tests the absence of {{{ parameter }}} setting in the {{{ path }}} file"
+  {{%- else %}}
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="tests the value of {{{ parameter }}} setting in the {{{ path }}} file"
+  {{%- endif %}}
+  id="test_{{{ rule_id }}}{{{ none_exist }}}" version="1">
+  <ind:object object_ref="obj_{{{ rule_id }}}{{{ none_exist }}}" />
+  {{%- if not none_exist %}}
+  <ind:state state_ref="state_{{{ rule_id }}}" />
+  {{%- endif %}}
+  </ind:textfilecontent54_test>
+{{%- endmacro %}}
+
+{{%- macro oval_line_in_file_object(path='', prefix_regex='^[\s]*', parameter='', separator_regex='[\s]+', value='', none_exist='') -%}}
+  <ind:textfilecontent54_object id="obj_{{{ rule_id }}}{{{ none_exist }}}" version="1">
+    <ind:filepath>{{{ path }}}</ind:filepath>
+    {{%- if none_exist %}}
+    <ind:pattern operation="pattern match">{{{ prefix_regex }}}{{{ parameter }}}{{{ separator_regex }}}</ind:pattern>
+    {{%- else %}}
+    <ind:pattern operation="pattern match">{{{ prefix_regex }}}{{{ parameter }}}{{{ separator_regex }}}(.*)[\s]*(?:|(?:#.*))?$</ind:pattern>
```

Comment:
Most of the time around separator can be spaces, so add next to `{{{ separator_regex }}}` `\s*` or maybe parameter, so user can choose if he wants them (and default `\s*`).

Reply:
I'm not sure if I get it. But the user can specify the `separator_regex` which means he can do something like `separator_regex='\s*=\s*'`. Wouldn't this be enough to cover this case?

Reply:
Yeah, right. When I saw it for the first time it came to my mind that I'd input `=` as separator and it wouldn't work. But because it's regex there's no problem with it.

It would be bit more convenient, if the user could input only separator and doesn't think about whitespace characters around, but it could cause problem with files, where is forbidden to have whitespaces.